### DISCUSSION TITLE
Sobre o await

### DISCUSSION THREAD

**Discussion by GeovaniDarcie** [balta-io/forum#348] [Tira Dúvidas] [answered]
Olá tudo bem? 

 Foi falado no curso que mesmo com o await, o código continua executando, se não depender da tarefa no await, porém, testei
aqui e o código é executado de foema sequencial, exemplo:
 
 await PrepararBatata();
Console.WriteLine("Nova etapa");

nesse caso rodando, ele executa tudo de prepararBatata primeiro, dentro desse método inclusive tem um tempo:
static async Task PrepararBatata()
{
    Console.WriteLine("1 - Preparando batatas");
    Console.WriteLine("-----------------------");
    await Task.Delay(8000);
    Console.WriteLine("Batatas prontas");
}

O resultado é sequencial, inclusive se eu ter outros métodos e chamo-los com o await, o print deles será sequencial, ex:
await PrepararBatata();
await carne =  PrepararCarne();
await mundo = InteragirComOmundo()

Pra ficar assincrono, eu teria que pegar as tasks e só depois dar o await

Task batatas = PrepararBatata();
Task carne =  PrepararCarne();
Task mundo = InteragirComOmundo();

await batatas;
await carne;
await mundo;

Os métodos que terminam primeiro, por exemplo, a carne fica pronta mais rápido, dai aparece primeiro a mensagem dela (mesmo eu chamando ela depois) depois o mundo, etc. Resultado final vai ficar assim o print: carne --> mundo --> batatas

mesmo eu chamando em outra ordem.

Li em vários lugares que o await aguarda sim, e não executa mais nada.
Mas no curso foi dito diferente...
Desculpa pelo tamanho da mensagem, espero que tenha entendido hehe.

Resumindo: o await faz aguardar o código e não executa nada, se eu quiser deixar assincrono devo chamar como task e depois dar await (ou usar aquele método Task.await())

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @GeovaniDarcie, como vai?

A palavra `await` significa aguardar, então neste caso o código para durante a execução deste método, ele não continua.
Continuaria caso você não utilizasse o `await`, somente chamando um método assíncrono.

**Reply by GeovaniDarcie** [reply]
Sim, isso mesmo, é que no curso você diz que mesmo com await ele continua a executar o código de forma assincrona

**Reply by andrebaltieri** [reply]
@GeovaniDarcie então desculpa, falei errado no curso, vou revisar, OK?
